search/searcher/search_geoshape.go

```diff
@@ -70,6 +70,20 @@ func buildRelationFilterOnShapes(ctx context.Context, dvReader index.DocValueRea
 	var dvShapeValue []byte
 	var startReading, finishReading bool
 	var reader *bytes.Reader
+
+	bufPool := make([][]byte, 11)
```

Comment:
We are creating a buffer pool per GeoShapeSearcher, 
But a Complex Query might end up creating multiple GeoShapeSearchers per query, 
In that case we will create that many buffer pools.

Example:
disjunct of a geoShape Query with a {conjunct of a geoShape Query AND match query } 
will end up creating two GeoShapeSearchers.

We should think about a way to create pool Per-Query and not Per-GeoShapeSearchers

Reply:
In future (may be phase-2), 
We can create a concurrent safe Dynamic Buffer pool per Bleve Import.

Pool Structure:
Pool: 
    buffers of size1
    buffers of size2
    ....
    
list of buffers of a specific size can be dynamic and can grow/shrink based on the exponential weighted moving average of usage metric of it.

Reply:
a suggestion over here is perhaps use the `ctx context.Context` that's been passed to the function. What I mean is if in index_impl.go where you're defining the single searcher (which internally can have multiple child searchers which are geo searchers) you pass set a callback function which returns a reference to the buffer pool, then the child searchers when they want to get a certain sized buffer from the pool can consult the common pool to grab the required buffer size. Furthermore, things are not really parallel in terms of Next() calls etc. so I don't think there can be a racy situation keeping that mind (please do double check it out though). As an example, you can take a look as to how `search.SearchIOStatsCallbackKey` is set and used throughout the queries etc. as a reference and then use something similar for the buffer pool stuff.

Reply:
For now I have left the creation of the buffer pool in the searcher, I will move it to the context once I test out if the calls are parallel and figure out how to implement it.